### PR TITLE
Improve Gui and Ged includes

### PR DESCRIPTION
**Pull request by linev**
1. Use `=delete` specifier for non-implemented methods
2. Use `nullptr` instead of 0
3. Reduce some includes
4. Fix `TGTextBuffer` - copy/assign operators were wrong, just delete them
5. Merge LinkDef files together

### CODE REVIEW

File: gui/gui/inc/LinkDef.h [hunk 1]
```diff
@@ -1,3 +1,290 @@
-#include "LinkDef1.h"
-#include "LinkDef2.h"
-#include "LinkDef3.h"
```
Comment (bellenot) [inline gui/gui/inc/LinkDef.h:3]: I think that was needed on Windows, due to some string/buffer/file limitations...

Comment (linev) [reply gui/gui/inc/LinkDef.h:3]: Also with cling?

Comment (bellenot) [reply gui/gui/inc/LinkDef.h:3]: Not sure... We'll see

Comment (linev) [reply gui/gui/inc/LinkDef.h:3]: As one can see from file history, before three separate GUI dictionaries were build.
And then in 2013 they were merged in one - while cling does not like to have many dictionaries for same library.

Comment (bellenot) [reply gui/gui/inc/LinkDef.h:3]: I think it was a source file (dictionary) size limitation with Visual Studio. Let's see if it's still the case...

Comment (linev) [reply gui/gui/inc/LinkDef.h:3]: But CLING generates one dictionary file anyway.

Comment (bellenot) [reply gui/gui/inc/LinkDef.h:3]: Well, if it compile AND works, then fine with me

**Review by bellenot — APPROVED** [review]
LGTM. Thanks Sergey!

**Pull request merged**